### PR TITLE
proto: Ensure input to EncodeFixed32 does not exceed max uint32

### PR DESCRIPTION
**Pull request by jonathangold**
See https://github.com/gogo/protobuf/issues/536

### CODE REVIEW

**Comment by jonathangold** [comment]
I signed it!

File: proto/encode.go [hunk 1]
```diff
@@ -139,6 +144,9 @@ func (p *Buffer) EncodeFixed64(x uint64) error {
 // This is the format for the
 // fixed32, sfixed32, and float protocol buffer types.
 func (p *Buffer) EncodeFixed32(x uint64) error {
+	if x > math.MaxUint32 {
```
Comment (puellanivis) [inline proto/encode.go]: https://play.golang.org/p/xSeqrvM8Bye 😟 I think the naïve use of this function will cause this change to break a lot of code if taken as is.

**Comment by dsnet** [comment]
The current behavior is unfortunate, but the reality is that people have come to depend on the overflow semantics.

**Comment by jonathangold** [comment]
@dsnet Fair enough. Thanks for having a look. Until next time!

**Pull request closed without merging**